connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -581,13 +582,23 @@ void incrementalAlterConfigs(Map<String, Config> topicConfigs) {
         }));
     }
 
-    private void updateTopicAcls(List<AclBinding> bindings) {
-        log.trace("Syncing {} topic ACL bindings.", bindings.size());
-        targetAdminClient.createAcls(bindings).values().forEach((k, v) -> v.whenComplete((x, e) -> {
-            if (e != null) {
-                log.warn("Could not sync ACL of topic {}.", k.pattern().name(), e);
-            }
-        }));
+    // Visible for testing
+    int updateTopicAcls(List<AclBinding> bindings) {
+        Set<AclBinding> addBindings = new HashSet<>(bindings);
+        addBindings.removeAll(knownTopicAclBindings);
+        int newBindCount = addBindings.size();
+        if (!addBindings.isEmpty()) {
+            log.info("Syncing new found {} topic ACL bindings.", newBindCount);
+            targetAdminClient.createAcls(addBindings).values().forEach((k, v) -> v.whenComplete((x, e) -> {
+                if (e != null) {
+                    log.warn("Could not sync ACL of topic {}.", k.pattern().name(), e);
+                }
+            }));
+            knownTopicAclBindings = new HashSet<>(bindings);
```

Comment:
Hmmm... won't this cause issues if the call above to `targetAdminClient::createAcls` fails?

Reply:
nice cacth !